com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,204 @@
+        // Check if there is an already existing banner layout. If so, remove it. Otherwise,
+        // create a new one.
+        bannerLayout?.let {
+            it.removeAllViews()
+            val bannerParent = it.parent as ViewGroup
+            bannerParent.removeView(it)
+        } ?: run {
+            bannerLayout = RelativeLayout(activity)
+            bannerLayout?.setBackgroundColor(Color.TRANSPARENT)
+        }
+
+        /*
+            //     TopLeft = 0,
+            //     TopCenter = 1,
+            //     TopRight = 2,
+            //     Center = 3,
+            //     BottomLeft = 4,
+            //     BottomCenter = 5,
+            //     BottomRight = 6
+        */
+        var bannerGravityPosition = 0
+        when (screenLocation) {
+            0 -> bannerGravityPosition = Gravity.TOP or Gravity.LEFT
+            1 -> bannerGravityPosition = Gravity.TOP or Gravity.CENTER_HORIZONTAL
+            2 -> bannerGravityPosition = Gravity.TOP or Gravity.RIGHT
+            3 -> bannerGravityPosition = Gravity.CENTER_VERTICAL or Gravity.CENTER_HORIZONTAL
+            4 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.LEFT
+            5 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.CENTER_HORIZONTAL
+            6 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.RIGHT
+        }
+        bannerLayout?.gravity = bannerGravityPosition
+
+        // Attach the banner layout to the activity.
+        val density = displayDensity
+        try {
+            when (ad.getSize() ?: HeliumBannerSize.STANDARD) {
+                HeliumBannerSize.LEADERBOARD -> ad.layoutParams = density?.let { getBannerLayoutParams(it, LEADERBOARD.first, LEADERBOARD.second) }
+                HeliumBannerSize.MEDIUM -> ad.layoutParams = density?.let { getBannerLayoutParams(it, MEDIUM.first, MEDIUM.second) }
+                HeliumBannerSize.STANDARD -> ad.layoutParams = density?.let { getBannerLayoutParams(it, STANDARD.first, STANDARD.second) }
+            }
+
+            // Attach the banner to the banner layout.
+            bannerLayout?.addView(ad)
+            activity?.addContentView(bannerLayout, ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT))
+
+            // This immediately sets the visibility of this banner. If this doesn't happen
+            // here, it is impossible to set the visibility later.
+            ad.visibility = View.VISIBLE
+
+            // This affects future visibility of the banner layout. Despite it never being
+            // set invisible, not setting this to visible here makes the banner not visible.
+            bannerLayout?.visibility = View.VISIBLE
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling banner load() - ${ex.message}")
+        }
+    }
+
+    private fun destroyBannerLayout() {
+        if (ad is HeliumBannerAd && bannerLayout != null) {
```

Comment:
if you want, you can probably just 
```
bannerLayout?.removeAllViews()
bannerLayout?.visibility = View.GONE
```
and not do the checks since you don't intend to call it if it's not a banner and even if it isn't, it won't do anything since bannerLayout will be null anyway. Also, there's no point checking if it's null or not since it won't do anything if it is null.

Reply:
True!